File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size()-1;
        int maxArea=0,left=0,right=n;

        while(left<right)
        {
            int currArea = n*min(height[left],height[right]);
            maxArea = max(maxArea,currArea);

            if(height[left]>height[right]){
                right--;
            }
            else{
                left++;
            }
            n--;
        }

        return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        size = len(height)
        left,right=0,size-1

        maxarea=0
        while left<right:
            size=size-1
            area = min(height[left],height[right])*size
            maxarea=max(maxarea,area)
            if height[left]<height[right]:
                left+=1
            else:
                right-=1

        return maxarea
        